src/middlewared/middlewared/plugins/nfs.py

```diff
@@ -54,6 +54,15 @@ def nfs_compress(self, nfs):
 
         verrors = ValidationErrors()
 
+        if new["v4"] and not await self.middleware.call("system.is_freenas"):
```

Comment:
@themylogin Looks great! I only have one problem.

We should only require this check if the user has enabled NFSv4 and checks the "Require Kerberos for NFSv4:" in the services section of the webui which is the `nfs['v4_krb']` in db.

We should change the ValidationError verbiage to say "Enabling kerberos authentication on TrueNAS HA requires setting the virtual hostname and domain.